apps/system/js/quick_settings.js

```diff
@@ -170,32 +170,19 @@
      * @memberof QuickSettings.prototype
      */
     monitorBluetoothChange: function() {
-      var self = this;
-      var btFirstSet = true;
-      SettingsListener.observe('bluetooth.enabled', true, function(value) {
-        // check self.bluetooth.dataset.enabled and value are identical
-        if ((self.bluetooth.dataset.enabled && value) ||
-          (self.bluetooth.dataset.enabled === undefined && !value)) {
-          return;
-        }
-
+      var req = navigator.mozSettings.createLock().get('bluetooth.enabled');
```

Comment:
We still need set the default state to quick settings button, which will not trigger the event.